### DISCUSSION TITLE
How to create polar plots with Csharp

### DISCUSSION THREAD

**Discussion by jackthomas818** [plotly/Plotly.NET#432] [Q&A] [answered]
Is there any support for producing polar plots with C# using Plotly.NET? The documentation only has F# code.

**Comment by kMutagene** [comment]
Hi @jackthomas818, yes there are multiple ways:

- F# is part of .NET, same as C#. You can use F# libraries from C#, meaning you can translate the samples.
- There is `Plotly.NET.CSharp` (see also https://plotly.net/#Installation), a more idiomatic C# API. It is written in C# but less feature complete than the core F# API. It has polar chart creation but less styling options:

  ```csharp
  using Plotly.NET.CSharp;
  Chart.PointPolar<int, int, string>(
        theta: new[] { 0, 45, 90, 180, 270, 360 },
        r: new[] { 1, 2, 3, 4, 5, 6 }
    );
  ```

![image](https://github.com/plotly/Plotly.NET/assets/21338071/cd344e0d-df0a-43f4-80c1-564cfcbd5ae9)

**Comment by jackthomas818** [comment]
I've got the basic polar plot displaying. Is it possible to rotate the polar plot clockwise 90 degrees so that the 0 degree is on the bottom of chart?

**Comment by kMutagene** [comment, marked as answer]
yes it is, the respective member is however on the `AngularAxis` object which is not wrapped in the CSharp lib. You can fallback on mixing both APIs, but will have to prefix the calls:

```fsharp
using Plotly.NET.CSharp;
using Plotly.NET;
using Plotly.NET.LayoutObjects;

var c = 
    Plotly.NET.CSharp.Chart.PointPolar<int, int, string>(
        theta: new[] { 0, 45, 90, 180, 270, 360 },
        r: new[] { 1, 2, 3, 4, 5, 6 }
    );

Plotly.NET.Chart.WithAngularAxis(
    AngularAxis.init<int, int, int, int, int, int>(Rotation: 270)
).Invoke(c)
```

![image](https://github.com/plotly/Plotly.NET/assets/21338071/8f48dd40-823c-4c00-860a-4e2ca4b31eeb)

I'll go ahead and close this, please use either the [Github discussions](https://github.com/plotly/Plotly.NET/discussions) tab or the [discord](https://discord.gg/k3kUtFY8DB) for questions - issues are more intended for bugs and feature requests
